ofrak_core/ofrak/core/patch_maker/modifiers.py

```diff
@@ -212,8 +214,20 @@ async def modify(self, resource: Resource, config: SegmentInjectorModifierConfig
 
             injection_tasks.append((region.resource, BinaryInjectorModifierConfig(patches)))
 
+        all_decendants = await resource.get_descendants()
         for injected_resource, injection_config in injection_tasks:
-            await injected_resource.run(BinaryInjectorModifier, injection_config)
+            result = await injected_resource.run(BinaryInjectorModifier, injection_config)
+            parents_to_delete = list(
+                filter(lambda r: r.get_id() in result.resources_modified, all_decendants)
+            )
+            to_delete = []
+            for parent in parents_to_delete:
+                to_delete.extend(
+                    list(
+                        await parent.get_descendants(r_filter=ResourceFilter.with_tags(Instruction))
```

Comment:
Shouldn't this also delete `BasicBlock`s at the very least and possibly `ComplexBlock`s as well? Do we even need any filtering here? I feel like anything that is wholly modified should have all of its children deleted.

Reply:
Makes sense, I removed the resource filter as I think it makes sense to delete all the children as well.